apps/omg_utils/lib/omg_utils/metrics.ex

```diff
@@ -33,18 +33,61 @@ defmodule OMG.Utils.Metrics do
   def measure_start(body, context) do
     # NOTE: the namespace and event group naming convention here is tentative.
     # It is possible we'll revert to standard coarser division into `web` and `background` namespaces Appsignal suggests
-    namespace = context.module |> Module.split() |> List.last() |> String.to_existing_atom()
-
-    if Enum.find(@discard_namespace_metrics, &match?(^&1, namespace)),
+    if Enum.find(@discard_namespace_metrics, &match?(^&1, namespace(context))),
       do: body,
-      else: Appsignal.Instrumentation.Decorators.transaction(namespace, body, context)
+      else: start_trace(body, context)
   end
 
   def measure_event(body, context) do
-    event_group = context.module |> Module.split() |> List.last() |> String.to_existing_atom()
-
-    if Enum.find(@discard_namespace_metrics, &match?(^&1, event_group)),
+    if Enum.find(@discard_namespace_metrics, &match?(^&1, namespace(context))),
       do: body,
-      else: Appsignal.Instrumentation.Decorators.transaction_event(event_group, body, context)
+      else: start_span(body, context)
+  end
+
+  defp start_trace(body, context) do
+    trace_name = trace_name(context)
+
+    quote do
+      _ = unquote(Tracer).start_trace(unquote(trace_name))
```

Comment:
I decided to match `_` instead of `{:ok, _}` here because I don't think we want to change any behavior if the tracer returns an error result... we want to continue as normal.